File: Luogu/Greedy/P1209.py
# P1209.py

arr = []
in_str = list(input().split(' '))
for str in in_str:
    if ('' != str):
        arr.append(int(str))
n = arr[0]
m = arr[1]

arr = []
for i in range(m):
    milk = []
    in_str = list(input().split(' '))
    for str in in_str:
        if ('' != str):
            milk.append(int(str))
    arr.append(milk)

arr.sort(key=lambda x: x[0], reverse=False)

sum = 0
for i in range(m):
    if (n <= 0):
        break
    if (n - arr[i][1] >= 0):
        sum += arr[i][0] * arr[i][1]
    else:
        sum += arr[i][0] * n
    n -= arr[i][1]

print(sum)

File: Luogu/Greedy/P1209.cc
#include <iostream>
#include <vector>
#include <algorithm>

struct milk {
    int p_, a_;

    milk() : p_(0), a_(0) {}
    ~milk() {}
};

bool cmp(const milk& a, const milk& b);

int main(void) {
    int n = 0, m = 0;
    int sum = 0;
    std::cin >> n >> m;

    std::vector<milk> arr(m, milk());

    for (int i = 0; i < m; i ++) {
        std::cin >> arr[i].p_ >> arr[i].a_;
    }

    std::sort(arr.begin(), arr.end(), cmp);

    for (int i = 0; i < m && n > 0; i ++) {
        if (n - arr[i].a_ >= 0) {
            sum += arr[i].a_ * arr[i].p_;
        } else {
            sum += arr[i].p_ * n;
        }
        n -= arr[i].a_;
    }

    std::cout << sum << std::endl;

    return 0;
}

bool cmp(const milk& a, const milk& b) {
    return a.p_ < b.p_;
}